applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C

```diff
@@ -0,0 +1,225 @@
+{
+namespace refinementControllers
+{
+    defineTypeNameAndDebug(uniformIntervals, 0);
+    addToRunTimeSelectionTable(refinementController, uniformIntervals, dictionary);
+}
+}
+
+// * * * * * * * * * * * * * * * * Constructors  * * * * * * * * * * * * * * //
+
+Foam::refinementControllers::uniformIntervals::uniformIntervals
+(
+    const PtrList<heatSourceModel>& sources,
+    const dictionary& dict,
+    const fvMesh& mesh
+)
+:
+    refinementController(typeName, sources, dict, mesh),
+    
+    coeffs_(refinementDict_.optionalSubDict(typeName + "Coeffs")),
+    intervals_(coeffs_.lookup<int>("intervals")),
+    dt_(mesh_.time().endTime().value() / intervals_),
+    updateTime_(0.0)
+{
+}
+
+
+// * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * * //
+
+bool Foam::refinementControllers::uniformIntervals::update()
+{
+    //- Call base class update
+    bool baseUpdate = refinementController::update();
+    
+    //- Update refinement field at update time
+    if (mesh_.time().value() >= updateTime_)
+    {
+        //- Update refinement index
+        lastRefinementIndex_ = mesh_.time().timeIndex();
+        
+        //- Call an extra mesh refinement on the first time
+        if (mesh_.time().timeIndex() == 0)
+        {
+            ++lastRefinementIndex_;
+        }
+        
+        //- Initialize refinement marker field in base class
+        refinementController::initializeRefinementField();
+    
+        //- Update next refinement time
+        const scalar currTime = mesh_.time().value();
+        updateTime_ = currTime + dt_;
+        
+        //- Buffer for bounding box calculations
+        const vector extend = 1e-10 * vector::one;
+        
+        const pointField& points = mesh_.points();
+        
+        //- Update refinement marker field
+        forAll(sources_, i)
+        {        
+            //- Create dummy time for forward march scheme
+            scalar beamTime = currTime;
+            
+            //- March along beam path until next update time is reached
+            while (beamTime < updateTime_)
+            {
+                //- Check if path is still active
+                if (!sources_[i].beam().activePath(beamTime))
+                {
+                    break;
+                }
+                
+                //- Mark cells near beam at current time/location for
+                //  refinement
+                vector beamDims = sources_[i].dimensions();
+                vector currPos = sources_[i].beam().position(beamTime);
+                treeBoundBox beamBb(currPos - beamDims, currPos + beamDims);
+                
+                forAll(mesh_.C(), celli)
+                {
+                    //- Don't redo checks if cell is already marked
+                    if (refinementField_[celli] > 0.0)
+                    {
+                        continue;
+                    }
+                    
+                    treeBoundBox cellBb(point::max, point::min);
```

Comment:
For the case of multiple beams, it may be more performant to handle the calculation of the cell bounding boxes outside the forAll(sources_, i) loop. Lets make sure we don't have any additional repeated logic here in future PRs

Reply:
Opened issue #15 for this